### PR TITLE
add func to check if domain is a member of given domain

### PR DESCRIPTION
**Pull request by wesinator**
opted for using other d8s functions ~, was easier than I thought compared to implementing algo from scratch (especially without regex).~

if this doesn't make sense as a function in here, feel free to reject.

### CODE REVIEW

**Comment by fhightower** [comment]
Thanks for the PR @wesinator! I hope to take a look at it in the next few days. Thanks for contributing 🎉

File: tests/test_domains.py [hunk 1]
```diff
@@ -75,6 +76,22 @@ def test_domain_as_unicode_docs_1():
...
+    assert domain_is_member('test1.example.com', 'test.example.com') is False
+    assert domain_is_member('example.com.notdomain.com', 'example.com') is False
+    assert domain_is_member('test.example.com.notdomain.com', 'example.com') is False
+    assert domain_is_member('1.test.example.com.notdomain.com', 'test.example.com') is False
+    assert domain_is_member('test.example.com', 'test.example.com') is True
+    assert domain_is_member('subdomain.test.example.com', 'test.example.com') is True
+    assert domain_is_member('example.com', 'testexample.com') is False
+    assert domain_is_member('example.com', 'test.testexample.com') is False
+    assert domain_is_member('test.example.com', 'test.testexample.com') is False
+    assert domain_is_member('example.com', 'example.com.br') is False
```
Comment (fhightower) [inline tests/test_domains.py]: Thanks for adding the tests!

File: d8s_domains/domains.py [hunk 2]
```diff
@@ -86,6 +86,33 @@ def domain_rank(domain_name: str) -> int:
...
+        base_subdomains = base_subdomain_str.split('.')
+        base_last_subdomain = base_subdomains[-1]
+
+        if not base_last_subdomain:
+            return domain_to_check.endswith("." + domain_base)
+        elif last_subdomain == base_last_subdomain:
+            return True
+    return False
+
+
```
Comment (fhightower) [inline d8s_domains/domains.py:115]: Thanks for suggesting this function! I do have a question. From what I can tell, the body of this function could be simplified to:

```python
return domain_to_check.endswith(domain_base)
```

and it would still do everything you want. Does that make sense? Are there edge-cases this change would not handle?

For example, the code snippet below shows a more simple version of the `domain_is_member` function that passes all of your test cases:

```python
def domain_is_member(domain_to_check, domain_base):
    return domain_to_check.endswith(domain_base)


assert domain_is_member('example.com', 'example.com') is True
assert domain_is_member('example.com', 'google.com') is False
assert domain_is_member('test.example.com', 'example.com') is True
assert domain_is_member('example.com', 'test.example.com') is False
assert domain_is_member('t.example.com', 'test.example.com') is False
assert domain_is_member('test1.example.com', 'test.example.com') is False
assert domain_is_member('example.com.notdomain.com', 'example.com') is False
assert domain_is_member('test.example.com.notdomain.com', 'example.com') is False
assert domain_is_member('1.test.example.com.notdomain.com', 'test.example.com') is False
assert domain_is_member('test.example.com', 'test.example.com') is True
assert domain_is_member('subdomain.test.example.com', 'test.example.com') is True
assert domain_is_member('example.com', 'testexample.com') is False
assert domain_is_member('example.com', 'test.testexample.com') is False
assert domain_is_member('test.example.com', 'test.testexample.com') is False
assert domain_is_member('example.com', 'example.com.br') is False
```

Is there an advantage to doing it the way you suggested?

**Review by fhightower — COMMENTED** [review]
Thanks for submitting this PR @wesinator! I do have a question (posted below).

File: d8s_domains/domains.py (hunk 2 above)
Comment (wesinator) [reply d8s_domains/domains.py:115]: Yes, `assert domain_is_member('testexample.com', 'example.com') is False` will not pass with a simple endswith 

The purpose of this function is to check if the domain is a subdomain or the exact domain.

If it were that simple I would've done it that way
:)

Comment (wesinator) [reply d8s_domains/domains.py:115]: some tests were not in the correct order. I'll add more tests for clarity

Comment (fhightower) [reply d8s_domains/domains.py:115]: Haha, I figured I was missing something... thanks for pointing that out.

**Review by fhightower — APPROVED** [review]
LGTM 🚀 

Thanks for the contribution @wesinator 🎉
